Assets/MixedRealityToolkit/Utilities/Editor/DependencyWindow.cs

```diff
@@ -0,0 +1,339 @@
+                    foreach (var dependency in dependencies)
+                    {
+                        node.assetsThisDependsOn.Add(EnsureNode(dependency));
+                        EnsureNode(dependency).assetsDependentOnThis.Add(node);
+                    }
+                }
+            }
+
+            EditorUtility.ClearProgressBar();
+
+            assetGraphRefreshTime = (float)((DateTime.UtcNow - beginTime).TotalSeconds);
+        }
+
+        private AssetGraphNode EnsureNode(string guid)
+        {
+            AssetGraphNode node;
+
+            if (!dependencyGraph.TryGetValue(guid, out node))
+            {
+                node = new AssetGraphNode();
+                node.guid = guid;
+                dependencyGraph.Add(guid, node);
+            }
+
+            return node;
+        }
+
+        private static bool IsAsset(string file)
+        {
+            return File.Exists(file);
+        }
+
+        private static bool IsGuidValid(string guid)
+        {
+            return !string.IsNullOrEmpty(guid) && guid != nullGuid;
+        }
+
+        private static string GetGuidFromMeta(string file)
+        {
+            try
+            {
+                string[] lines = File.ReadAllLines(file);
+
+                foreach (var line in lines)
+                {
+                    if (line.StartsWith(guidPrefix))
+                    {
+                        return line.Split(' ')[1];
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+            }
+
+            return null;
+        }
+
+        private static List<string> GetDependenciesInFile(string file)
+        {
+            var guids = new List<string>();
+
+            try
+            {
+                foreach (var line in File.ReadAllLines(file))
+                {
+                    var index = line.IndexOf(guidPrefix);
+
+                    if (index > 0)
+                    {
+                        var guid = line.Substring(index + guidPrefix.Length, guidCharacterCount);
+
+                        if (IsGuidValid(guid) && !guids.Contains(guid))
+                        {
+                            guids.Add(guid);
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+            }
+
+            return guids;
+        }
+
+        private static void DrawDependency(AssetGraphNode node, int depth, int maxDepth)
+        {
+            EditorGUILayout.BeginHorizontal();
+            {
+                GUILayout.Space(depth * 16);
+
+                if (depth != maxDepth)
+                {
+                    var path = AssetDatabase.GUIDToAssetPath(node.guid);
+
+                    if (GUILayout.Button("Select", GUILayout.ExpandWidth(false)))
```

Comment:
Does EditorGUIlayout.ObjectField not work? Like you use earlier for AssetSelection. Or do you like showing the full path for the asset?
assetSelection = EditorGUILayout.ObjectField("Asset Selection", assetSelection, typeof(Object), false);

Reply:
I didn't even think of this. Great suggestion! Switched it over.